File: generate_pdf.swift
import Foundation
import AppKit
import CoreGraphics
import CoreText

let filePath = "/Users/giyoshimiken/Documents/Flow_1/Sample_Document.pdf"
let url = URL(fileURLWithPath: filePath)

var mediaBox = CGRect(x: 0, y: 0, width: 595.2, height: 841.8) // A4
guard let context = CGContext(url as CFURL, mediaBox: &mediaBox, nil) else {
    print("Failed to create context")
    exit(1)
}

context.beginPDFPage(nil)

// Draw Background
context.setFillColor(gray: 0.98, alpha: 1.0)
context.fill(mediaBox)

// Draw Title
let title = "Make it Flow - Sample Test Document"
let titleAttributes: [NSAttributedString.Key: Any] = [
    .font: NSFont.boldSystemFont(ofSize: 28),
    .foregroundColor: NSColor.black
]
let titleString = NSAttributedString(string: title, attributes: titleAttributes)
let titleLine = CTLineCreateWithAttributedString(titleString)
context.textPosition = CGPoint(x: 50, y: 841.8 - 100)
CTLineDraw(titleLine, context)

// Draw Body
let body = """
Welcome to Make it Flow!

This is a generated test PDF document for the Flow application. 
Make it Flow is designed to take any complex PDF and extract its text beautifully, 
preserving the original reading order, detecting headers, paragraphs, and even 
bypassing messy multi-column layouts using advanced Apple Vision frameworks.

Here are some features you can test with this document:
1. Text extraction accuracy
2. Sentence boundary repair using local LLM
3. EPUB compilation speed and formatting

If you can read this in your EPUB, the conversion was a huge success! 
Enjoy testing your amazing new app.
"""

let bodyAttributes: [NSAttributedString.Key: Any] = [
    .font: NSFont.systemFont(ofSize: 16),
    .foregroundColor: NSColor.darkGray
]
let bodyString = NSAttributedString(string: body, attributes: bodyAttributes)
let framesetter = CTFramesetterCreateWithAttributedString(bodyString)
let textRect = CGRect(x: 50, y: 100, width: 495.2, height: 600)
let path = CGPath(rect: textRect, transform: nil)
let frame = CTFramesetterCreateFrame(framesetter, CFRangeMake(0, 0), path, nil)

context.saveGState()
CTFrameDraw(frame, context)
context.restoreGState()

context.endPDFPage()
context.closePDF()

print("PDF generated successfully at: \\(filePath)")

File: generate_pdf.py
import sys
import CoreGraphics as CG
import CoreText as CT
import Foundation as NS

def create_pdf(file_path):
    # PDF dimensions (A4 size)
    width, height = 595.2, 841.8
    page_rect = CG.CGRectMake(0, 0, width, height)

    url = NS.NSURL.fileURLWithPath_(file_path)
    context = CG.CGPDFContextCreateWithURL(url, page_rect, None)

    CG.CGPDFContextBeginPage(context, None)
    
    # Draw background
    CG.CGContextSetRGBFillColor(context, 0.98, 0.98, 0.98, 1.0)
    CG.CGContextFillRect(context, page_rect)

    # Title
    title = "Make it Flow - Sample Document"
    title_attr = {
        NS.NSFontAttributeName: NS.NSFont.systemFontOfSize_weight_(28.0, 0.5), # Bold
        NS.NSForegroundColorAttributeName: NS.NSColor.blackColor()
    }
    title_str = NS.NSAttributedString.alloc().initWithString_attributes_(title, title_attr)
    title_line = CT.CTLineCreateWithAttributedString(title_str)
    
    CG.CGContextSetTextPosition(context, 50, height - 100)
    CT.CTLineDraw(title_line, context)

    # Body text
    body = """
    Welcome to Make it Flow!

    This is a sample PDF document generated automatically to test the Flow application. 
    Make it Flow is designed to take any complex PDF and extract its text beautifully, 
    preserving the original reading order, detecting headers, paragraphs, and even 
    bypassing messy multi-column layouts using advanced Apple Vision frameworks.

    Here are some features you can test with this document:
    1. Text extraction accuracy
    2. Sentence boundary repair using local LLM
    3. EPUB compilation speed and formatting

    Enjoy testing your amazing new app!
    """
    
    body_attr = {
        NS.NSFontAttributeName: NS.NSFont.systemFontOfSize_(14.0),
        NS.NSForegroundColorAttributeName: NS.NSColor.darkGrayColor()
    }
    body_str = NS.NSAttributedString.alloc().initWithString_attributes_(body, body_attr)
    
    # Text box (using framesetter for multi-line)
    framesetter = CT.CTFramesetterCreateWithAttributedString(body_str)
    text_rect = CG.CGRectMake(50, 400, width - 100, height - 200)
    path = CG.CGPathCreateWithRect(text_rect, None)
    frame = CT.CTFramesetterCreateFrame(framesetter, CG.CFRangeMake(0, 0), path, None)

    CT.CTFrameDraw(frame, context)

    CG.CGPDFContextEndPage(context)
    CG.CGPDFContextClose(context)
    print(f"Created PDF at: {file_path}")

if __name__ == '__main__':
    create_pdf('Sample_Document.pdf')
